extensions/amp-analytics/0.1/variables.js

```diff
@@ -173,6 +174,17 @@ function matchMacro(string, matchPattern, opt_matchingGroupIndexStr) {
   return matches && matches[index] ? matches[index] : '';
 }
 
+/**
+ * Returns a comma separated list of active experiment branches or null
+ * if none are active.
+ * @param {!Window} win
+ * @return {?string}
+ */
+function experimentBranchesMacro(win) {
+  const branches = getActiveExperimentBranches(win);
+  return branches ? Object.values(branches).join(',') : null;
```

Comment:
hmmm what's the design for this. My understanding of the experiment branch looks like the following
exp1: branch1, branch2
exp2: branch1, branch2
Where the experiment name plus the branch name matters.

Reply:
I'm sort of mirroring what they are doing in the google ads code. A experiment usually looks like this https://github.com/ampproject/amphtml/blob/ce4569ee8d6561c59d625e1b8f84d542be84a8aa/extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js#L135-L140 where you are right, they have a name and n number of branches, but the branch ids should be unique (I guess we don't actually enforce uniqueness though). Anyway on google ads side they then just flatten the branches into a string of comma separated numbers https://github.com/ampproject/amphtml/blob/ce4569ee8d6561c59d625e1b8f84d542be84a8aa/ads/google/a4a/utils.js#L738
I think the idea is that this makes parsing the exp ids server side much easier as long as there are no collisions.

Reply:
If the idea is that the macro is only for google ad, then it's your call.
Please consider adding it to a4a not analytics. 
At lease leave a comment and remove the support of `${experimentBranches}`

Reply:
This could be used for non-ads related experiments as well. Anyone that wants to do a client side diverted experiment and report that value. Would you like me to just add the experiment name into the macro? Something like `expA:123,expB:456`. I think the tradeoff is 
- list of ids:
-- easier to parse server side
-- possibility of non-unique values
- names+ids
-- harder to parse
-- no collisions

I don't feel strongly either way.

Reply:
My recommendation is:
If this is going to be ads specific, then do whatever is easier
If you want to ship this as a more generic feature, please have some way to avoid potential conflict. Any 3P component can decide to introduce such an experiment and it would eventually be difficult to manage all branch ids. I'm fine with not including the experiment name if all experiments are declared at one place and we can enforce the uniqueness of the branch id through review process.

Reply:
You are right this is probably better future proofing. Enforcing uniqueness would be a larger change but probably also a good idea. Added exp names to the macro e.g. `expA:123,expB:456`